pkg/network/usm/dump.go

```diff
@@ -76,6 +66,10 @@ func dumpMapsHandler(manager *manager.Manager, mapName string, currentMap *ebpf.
 		for iter.Next(unsafe.Pointer(&key), unsafe.Pointer(&value)) {
 			output.WriteString(spew.Sdump(key, value))
 		}
+	default: // Go through enabled protocols in case one of them now how to handle the current map
```

Comment:
That would consume a bit more memory though. Also, every map is not dumped the same way because of the different types they hold.
I don't think map dumps are a frequent occurence, and we don't have so many maps and protocols that looping through them would be a performance problem that would require an optimisation.

Reply:
Memory increase should be negligible as this map will have a constant size (number of maps from all protocols). Also this map maps the eBPF map name to the dump function so it shouldn't be a problem to print all different eBPF maps. I agree that the performance is not a main concern here, I was aiming more for the code structure. Anyway that's only a suggestion so I'm resolving this comment